### PR TITLE
Clobber old device list stream entries

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/deviceinbox.py [hunk 1]
```diff
@@ -31,10 +31,10 @@ def __init__(self, hs):
...
 
         self.register_background_index_update(
-            "device_inbox_stream_index",
-            index_name="device_inbox_stream_id_user_id",
-            table="device_inbox",
-            columns=["stream_id", "user_id"],
+            "device_lists_stream_idx",
+            index_name="device_lists_stream_user_id",
+            table="device_lists_stream",
+            columns=["user_id", "device_id"],
```
Comment (NegativeMjark) [inline synapse/storage/deviceinbox.py]: Umm, What happened to the `(stream_id, user_id)` index?
Should the new index be called `device_lists_stream_user_id_device_id`?

File: synapse/storage/devices.py [hunk 2]
```diff
@@ -33,6 +33,13 @@ def __init__(self, hs):
             self._prune_old_outbound_device_pokes, 60 * 60 * 1000
         )
 
+        self.register_background_index_update(
+            "device_inbox_stream_index",
+            index_name="device_inbox_stream_id_user_id",
+            table="device_inbox",
```
Comment (NegativeMjark) [inline synapse/storage/devices.py]: Why isn't this in the device_inbox file anymore?

File: synapse/storage/deviceinbox.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/storage/deviceinbox.py]: Ha, I got confused after copy pasting...

File: synapse/storage/schema/delta/41/device_list_stream_idx.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+INSERT into background_updates (update_name, progress_json)
+    VALUES ('device_lists_stream_idx', '{}');
```
Comment (NegativeMjark) [inline synapse/storage/schema/delta/41/device_list_stream_idx.sql]: Is calling this `idx` or `index` more consistent with what came before?

Comment (erikjohnston) [inline synapse/storage/schema/delta/41/device_list_stream_idx.sql]: There's a mix

**Comment by erikjohnston** [comment]
@NegativeMjark PTAL

**Pull request merged**